### PR TITLE
goto_trace now contains the arguments of function calls

### PR DESCRIPTION
**Pull request by kroening**

### CODE REVIEW

**Comment by tautschnig** [comment]
I'm assuming the first commit is #2818 - reviewing the second one only.

**Review by tautschnig — COMMENTED** [review]
I'm inclined to treat this as work-in-progress? It looks good to me, but thus far is only increasing the memory footprint (adding a vector to each SSA step) without making use of the newly stored data. Could further work be added to this PR that a) makes use of it and b) explicitly tests (e.g., that arguments have undergone constant propagation) it?

**Comment by kroening** [comment]
I'll add a user.
An obvious idea would be to offer a --stack-trace option.

File: regression/cbmc/trace_show_function_calls/test.desc [hunk 1]
```diff
@@ -3,9 +3,9 @@ main.c
 --trace --trace-show-function-calls
 ^EXIT=10$
 ^SIGNAL=0$
-Function call: function_to_call \(depth 2\)
+Function call: function_to_call\(-2147483647\) \(depth 2\)
```
Comment (smowton) [inline regression/cbmc/trace_show_function_calls/test.desc:6]: Are these random nondet choices? If so they're probably not good to constrain in the test

**Review by smowton — COMMENTED** [review]
Pinging @thk123 as this will allow us to scrap `interpreter_input_synthesis`'s reconstruction of call sequences

File: regression/cbmc/trace_show_function_calls/test.desc (hunk 1 above)
Comment (tautschnig) [reply regression/cbmc/trace_show_function_calls/test.desc:6]: No, they seem to be hard-coded in the test. (I'm just wondering whether the values are portable across all architectures.)

**Review by tautschnig — APPROVED** [review]
Looks good to me!

**Pull request merged**